server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -427,6 +450,7 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeOptionalWriteable(shardId);
             out.writeOptionalString(readerCacheKeyId);
             out.writeBytesReference(value);
+            out.writeInt(indexShardHashCode);
```

Comment:
Does it ever make sense to serialize this `indexShardHashCode` to a `StreamOutput`?

The object identity is only going to be useful within the current process. We should never pass it between nodes.

Reply:
Yeah we need to serialize this as well for disk cache.
Once it is removed from the disk cache, we need to deserialize and use the same `hashcode` logic in `onRemoval` method.

Reply:
The `key` object is serialized only for the diskcache which happens to use the `Writable` interface for ser-de to read and write on disk here. Should we add some java doc for this class explaining the same. Otherwise it is confusing as to why `indexShardHashCode` is being serialized to StreamOutput which is primarily used for node-to-node communication. Is there any explicit way to fail if serialization is happening for node-to-node communication instead ? Probably in later versions we should move away from StreamInput/Output based ser-de as that is specifically for node to node communication.

Reply:
@sohami Good point. I will add a comment to avoid that confusion. This cache key is never passed between nodes, we used the default StreamOutput/input purely as a serializing logic.
We do have a plan to try out other serialization mechanism in follow up phases.